// @create-figma-plugin/ui/lib/components/layout/vertical-space/vertical-space.stories.js
import { h } from "preact";
import { VerticalSpace } from "./vertical-space.js";
export default {
    parameters: {
        fixedWidth: true
    },
    title: "Components/Layout/Vertical Space"
};
export const ExtraSmallSpace = function () {
    const style = { backgroundColor: "#daebf7" };
    return (h("div", { style: style }, h(VerticalSpace, { space: "extraSmall" })));
};
export const SmallSpace = function () {
    const style = { backgroundColor: "#daebf7" };
    return (h("div", { style: style }, h(VerticalSpace, { space: "small" })));
};
export const MediumSpace = function () {
    const style = { backgroundColor: "#daebf7" };
    return (h("div", { style: style }, h(VerticalSpace, { space: "medium" })));
};
export const LargeSpace = function () {
    const style = { backgroundColor: "#daebf7" };
    return (h("div", { style: style }, h(VerticalSpace, { space: "large" })));
};
export const ExtraLargeSpace = function () {
    const style = { backgroundColor: "#daebf7" };
    return (h("div", { style: style }, h(VerticalSpace, { space: "extraLarge" })));
};
