// @create-figma-plugin/ui/lib/components/textbox/textbox/textbox.stories.js
import { MIXED_STRING } from "@create-figma-plugin/utilities";
import { h } from "preact";
import { useState } from "preact/hooks";
import { useInitialFocus } from "../../../hooks/use-initial-focus/use-initial-focus.js";
import { IconLayerFrame16 } from "../../icon/icon-16/icon-layer-frame-16.js";
import { Textbox } from "./textbox.js";
export default {
    parameters: {
        fixedWidth: true
    },
    title: "Components/Textbox"
};
export const Empty = function () {
    const [value, setValue] = useState("");
    function handleInput(event) {
        const newValue = event.currentTarget.value;
        console.log(newValue);
        setValue(newValue);
    }
    return h(Textbox, { onInput: handleInput, value: value });
};
export const Focused = function () {
    const [value, setValue] = useState("");
    function handleInput(event) {
        const newValue = event.currentTarget.value;
        console.log(newValue);
        setValue(newValue);
    }
    return h(Textbox, { ...useInitialFocus(), onInput: handleInput, value: value });
};
export const Placeholder = function () {
    const [value, setValue] = useState("");
    function handleInput(event) {
        const newValue = event.currentTarget.value;
        console.log(newValue);
        setValue(newValue);
    }
    return (h(Textbox, { onInput: handleInput, placeholder: "Placeholder", value: value }));
};
export const Filled = function () {
    const [value, setValue] = useState("Text");
    function handleInput(event) {
        const newValue = event.currentTarget.value;
        console.log(newValue);
        setValue(newValue);
    }
    return h(Textbox, { onInput: handleInput, value: value });
};
export const Disabled = function () {
    function handleInput() {
        throw new Error("This function should not be called");
    }
    return h(Textbox, { disabled: true, onInput: handleInput, value: "Text" });
};
export const NoBorder = function () {
    const [value, setValue] = useState("Text");
    function handleInput(event) {
        const newValue = event.currentTarget.value;
        console.log(newValue);
        setValue(newValue);
    }
    return h(Textbox, { noBorder: true, onInput: handleInput, value: value });
};
export const Icon = function () {
    const [value, setValue] = useState("Text");
    function handleInput(event) {
        const newValue = event.currentTarget.value;
        console.log(newValue);
        setValue(newValue);
    }
    return (h(Textbox, { icon: h(IconLayerFrame16, null), onInput: handleInput, value: value }));
};
export const TextIcon = function () {
    const [value, setValue] = useState("Text");
    function handleInput(event) {
        const newValue = event.currentTarget.value;
        console.log(newValue);
        setValue(newValue);
    }
    return h(Textbox, { icon: "X", onInput: handleInput, value: value });
};
export const RevertOnEscapeKeyDown = function () {
    const [value, setValue] = useState("Text");
    function handleInput(event) {
        const newValue = event.currentTarget.value;
        console.log(newValue);
        setValue(newValue);
    }
    return h(Textbox, { onInput: handleInput, revertOnEscapeKeyDown: true, value: value });
};
export const ValidateOnBlur = function () {
    const [value, setValue] = useState("Text");
    function handleInput(event) {
        const newValue = event.currentTarget.value;
        console.log(newValue);
        setValue(newValue);
    }
    function validateOnBlur(value) {
        return value !== "";
    }
    return (h(Textbox, { onInput: handleInput, validateOnBlur: validateOnBlur, value: value }));
};
export const Mixed = function () {
    const [value, setValue] = useState(MIXED_STRING);
    function handleInput(event) {
        const newValue = event.currentTarget.value;
        console.log(newValue);
        setValue(newValue);
    }
    return h(Textbox, { onInput: handleInput, value: value });
};
export const Password = function () {
    const [value, setValue] = useState("Text");
    function handleInput(event) {
        const newValue = event.currentTarget.value;
        console.log(newValue);
        setValue(newValue);
    }
    return h(Textbox, { onInput: handleInput, password: true, value: value });
};
export const OnValueInput = function () {
    const [value, setValue] = useState("Text");
    return h(Textbox, { onValueInput: setValue, value: value });
};
