// @create-figma-plugin/ui/lib/components/tabs/tabs.stories.js
import { h } from "preact";
import { useState } from "preact/hooks";
import { Tabs } from "./tabs.js";
export default {
    parameters: {
        fixedWidth: true
    },
    title: "Components/Tabs"
};
export const Unselected = function () {
    const [value, setValue] = useState(null);
    const options = [
        { children: h("div", null, "Foo"), value: "foo" },
        { children: h("div", null, "Bar"), value: "bar" },
        { children: h("div", null, "Baz"), value: "baz" }
    ];
    function handleChange(event) {
        const newValue = event.currentTarget.value;
        console.log(newValue);
        setValue(newValue);
    }
    return h(Tabs, { onChange: handleChange, options: options, value: value });
};
export const Selected = function () {
    const [value, setValue] = useState("bar");
    const options = [
        { children: h("div", null, "Foo"), value: "foo" },
        { children: h("div", null, "Bar"), value: "bar" },
        { children: h("div", null, "Baz"), value: "baz" }
    ];
    function handleChange(event) {
        const newValue = event.currentTarget.value;
        console.log(newValue);
        setValue(newValue);
    }
    return h(Tabs, { onChange: handleChange, options: options, value: value });
};
export const OnValueChange = function () {
    const [value, setValue] = useState("bar");
    const options = [
        { children: h("div", null, "Foo"), value: "foo" },
        { children: h("div", null, "Bar"), value: "bar" },
        { children: h("div", null, "Baz"), value: "baz" }
    ];
    return h(Tabs, { onValueChange: setValue, options: options, value: value });
};
