// @create-figma-plugin/ui/lib/components/layout/inline/inline.stories.js
import { h } from "preact";
import { Inline } from "./inline.js";
export default { title: "Components/Layout/Inline" };
export const NoSpace = function () {
    const style = { backgroundColor: "#daebf7" };
    return (h(Inline, null, h("div", { style: style }, "foo"), h("div", { style: style }, "bar"), h("div", { style: style }, "baz")));
};
export const ExtraSmallSpace = function () {
    const style = { backgroundColor: "#daebf7" };
    return (h(Inline, { space: "extraSmall" }, h("div", { style: style }, "foo"), h("div", { style: style }, "bar"), h("div", { style: style }, "baz")));
};
export const SmallSpace = function () {
    const style = { backgroundColor: "#daebf7" };
    return (h(Inline, { space: "small" }, h("div", { style: style }, "foo"), h("div", { style: style }, "bar"), h("div", { style: style }, "baz")));
};
export const MediumSpace = function () {
    const style = { backgroundColor: "#daebf7" };
    return (h(Inline, { space: "medium" }, h("div", { style: style }, "foo"), h("div", { style: style }, "bar"), h("div", { style: style }, "baz")));
};
export const LargeSpace = function () {
    const style = { backgroundColor: "#daebf7" };
    return (h(Inline, { space: "large" }, h("div", { style: style }, "foo"), h("div", { style: style }, "bar"), h("div", { style: style }, "baz")));
};
export const ExtraLargeSpace = function () {
    const style = { backgroundColor: "#daebf7" };
    return (h(Inline, { space: "extraLarge" }, h("div", { style: style }, "foo"), h("div", { style: style }, "bar"), h("div", { style: style }, "baz")));
};
