// @create-figma-plugin/ui/lib/components/button/button.stories.js
import { h } from "preact";
import { useInitialFocus } from "../../hooks/use-initial-focus/use-initial-focus.js";
import { Button } from "./button.js";
export default { title: "Components/Button" };
export const Default = function () {
    function handleClick(event) {
        console.log(event);
    }
    return h(Button, { onClick: handleClick }, "Text");
};
export const Focused = function () {
    function handleClick(event) {
        console.log(event);
    }
    return (h(Button, { ...useInitialFocus(), onClick: handleClick }, "Text"));
};
export const Loading = function () {
    function handleClick() {
        throw new Error("This function should not be called");
    }
    return (h(Button, { loading: true, onClick: handleClick }, "Text"));
};
export const Disabled = function () {
    function handleClick() {
        throw new Error("This function should not be called");
    }
    return (h(Button, { disabled: true, onClick: handleClick }, "Text"));
};
export const FullWidth = function () {
    function handleClick(event) {
        console.log(event);
    }
    return (h(Button, { fullWidth: true, onClick: handleClick }, "Text"));
};
FullWidth.parameters = {
    fixedWidth: true
};
export const FullWidthLongText = function () {
    const longText = Array(100).fill("Text").join(" ");
    function handleClick(event) {
        console.log(event);
    }
    return (h(Button, { fullWidth: true, onClick: handleClick }, longText));
};
FullWidthLongText.parameters = {
    fixedWidth: true
};
export const Secondary = function () {
    function handleClick(event) {
        console.log(event);
    }
    return (h(Button, { onClick: handleClick, secondary: true }, "Text"));
};
export const Destructive = function () {
    function handleClick(event) {
        console.log(event);
    }
    return (h(Button, { destructive: true, onClick: handleClick }, "Text"));
};
export const DestructiveSecondary = function () {
    function handleClick(event) {
        console.log(event);
    }
    return (h(Button, { destructive: true, onClick: handleClick, secondary: true }, "Text"));
};
