// @create-figma-plugin/ui/lib/components/modal/modal.stories.js
import { Fragment, h } from "preact";
import { useState } from "preact/hooks";
import { Button } from "../button/button.js";
import { IconReturn32 } from "../icon/icon-32/icon-return-32.js";
import { Modal } from "./modal.js";
export default { title: "Components/Modal" };
export const Default = function () {
    const [isOpen, setIsOpen] = useState(false);
    function handleOpenButtonClick() {
        setIsOpen(true);
    }
    function handleCloseButtonClick() {
        setIsOpen(false);
    }
    const style = { height: "160px", padding: "12px", width: "240px" };
    return (h(Fragment, null, h(Button, { fullWidth: true, onClick: handleOpenButtonClick }, "Open"), h(Modal, { isOpen: isOpen }, h("div", { style: style }, h(Button, { onClick: handleCloseButtonClick, secondary: true }, "Close")))));
};
export const PositionBottom = function () {
    const [isOpen, setIsOpen] = useState(false);
    function handleOpenButtonClick() {
        setIsOpen(true);
    }
    function handleCloseButtonClick() {
        setIsOpen(false);
    }
    const style = { height: "160px", padding: "12px" };
    return (h(Fragment, null, h(Button, { fullWidth: true, onClick: handleOpenButtonClick }, "Open"), h(Modal, { isOpen: isOpen, position: "bottom" }, h("div", { style: style }, h(Button, { onClick: handleCloseButtonClick, secondary: true }, "Close")))));
};
export const PositionLeft = function () {
    const [isOpen, setIsOpen] = useState(false);
    function handleOpenButtonClick() {
        setIsOpen(true);
    }
    function handleCloseButtonClick() {
        setIsOpen(false);
    }
    const style = { padding: "12px", width: "240px" };
    return (h(Fragment, null, h(Button, { fullWidth: true, onClick: handleOpenButtonClick }, "Open"), h(Modal, { isOpen: isOpen, position: "left" }, h("div", { style: style }, h(Button, { onClick: handleCloseButtonClick, secondary: true }, "Close")))));
};
export const PositionRight = function () {
    const [isOpen, setIsOpen] = useState(false);
    function handleOpenButtonClick() {
        setIsOpen(true);
    }
    function handleCloseButtonClick() {
        setIsOpen(false);
    }
    const style = { padding: "12px", width: "240px" };
    return (h(Fragment, null, h(Button, { fullWidth: true, onClick: handleOpenButtonClick }, "Open"), h(Modal, { isOpen: isOpen, position: "right" }, h("div", { style: style }, h(Button, { onClick: handleCloseButtonClick, secondary: true }, "Close")))));
};
export const CloseOnOverlayClick = function () {
    const [isOpen, setIsOpen] = useState(false);
    function handleOpenButtonClick() {
        setIsOpen(true);
    }
    function handleOverlayClick() {
        setIsOpen(false);
    }
    const style = { height: "160px", padding: "12px", width: "240px" };
    return (h(Fragment, null, h(Button, { fullWidth: true, onClick: handleOpenButtonClick }, "Open"), h(Modal, { isOpen: isOpen, onOverlayClick: handleOverlayClick }, h("div", { style: style }, "foo"))));
};
export const CloseOnEscapeKeyDown = function () {
    const [isOpen, setIsOpen] = useState(false);
    function handleOpenButtonClick() {
        setIsOpen(true);
    }
    function handleEscapeKeyDown() {
        setIsOpen(false);
    }
    const style = { height: "160px", padding: "12px", width: "240px" };
    return (h(Fragment, null, h(Button, { fullWidth: true, onClick: handleOpenButtonClick }, "Open"), h(Modal, { isOpen: isOpen, onEscapeKeyDown: handleEscapeKeyDown }, h("div", { style: style }, "foo"))));
};
export const Title = function () {
    const [isOpen, setIsOpen] = useState(false);
    function handleOpenButtonClick() {
        setIsOpen(true);
    }
    function handleCloseButtonClick() {
        setIsOpen(false);
    }
    const style = { height: "160px", padding: "12px", width: "240px" };
    return (h(Fragment, null, h(Button, { fullWidth: true, onClick: handleOpenButtonClick }, "Open"), h(Modal, { isOpen: isOpen, title: "foo" }, h("div", { style: style }, h(Button, { onClick: handleCloseButtonClick, secondary: true }, "Close")))));
};
export const CloseButton = function () {
    const [isOpen, setIsOpen] = useState(false);
    function handleOpenButtonClick() {
        setIsOpen(true);
    }
    function handleCloseButtonClick() {
        setIsOpen(false);
    }
    const style = { height: "160px", padding: "12px", width: "240px" };
    return (h(Fragment, null, h(Button, { fullWidth: true, onClick: handleOpenButtonClick }, "Open"), h(Modal, { isOpen: isOpen, onCloseButtonClick: handleCloseButtonClick, title: "foo" }, h("div", { style: style }, "bar"))));
};
export const CloseButtonPosition = function () {
    const [isOpen, setIsOpen] = useState(false);
    function handleOpenButtonClick() {
        setIsOpen(true);
    }
    function handleCloseButtonClick() {
        setIsOpen(false);
    }
    const style = { height: "160px", padding: "12px", width: "240px" };
    return (h(Fragment, null, h(Button, { fullWidth: true, onClick: handleOpenButtonClick }, "Open"), h(Modal, { closeButtonPosition: "left", isOpen: isOpen, onCloseButtonClick: handleCloseButtonClick, title: "foo" }, h("div", { style: style }, "bar"))));
};
export const CloseButtonIcon = function () {
    const [isOpen, setIsOpen] = useState(false);
    function handleOpenButtonClick() {
        setIsOpen(true);
    }
    function handleCloseButtonClick() {
        setIsOpen(false);
    }
    const style = { height: "160px", padding: "12px", width: "240px" };
    return (h(Fragment, null, h(Button, { fullWidth: true, onClick: handleOpenButtonClick }, "Open"), h(Modal, { closeButtonIcon: h(IconReturn32, null), closeButtonPosition: "left", isOpen: isOpen, onCloseButtonClick: handleCloseButtonClick, title: "foo" }, h("div", { style: style }, "bar"))));
};
